### DISCUSSION TITLE
Nested layouts

### DISCUSSION THREAD

**Discussion by Daniel-Knights** [harttle/liquidjs#603] [Q&A] [answered]
Hi, is there a way to nest `layout` tags? For example, if you had a `html` layout, and a `popup` layout that you wanted to use inside:

```liquid
{% layout "html" %}

<main>
  {% layout "popup" %}
    <p>Popup content</p>
  {% layout "popup" %}
    <p>Other popup content</p>
</main>
```

As a workaround, I'm using `capture` for the popup content and passing it to a `render` tag, but this affects readability, and you can't nest other `render` tags within (from what I can tell):

```liquid
{% layout "html" %}

<main>
  {% capture content %}
    <p>Popup content</p>
  {% endcapture %}
  {% render "popup", content: content %}
  {% capture content %}
    <p>Other popup content</p>
  {% endcapture %}
  {% render "popup", content: content %}
</main>
```

For context, I'm coming from a Handlebars background, and I'm not using LiquidJS with Shopify (just plainly as a templating language), so maybe I'm just approaching this in the wrong way.

**Comment by harttle** [comment, marked as answer]
This is how liquid layout is designed, we need try our best to align with Shopify so we cannot change existing tags too much. One problem to achieve your use case is there’s no endlayout, so we don’t know where the last child layout tag ends.

There’s another approach: move two popups into its own file, and use the layout tag separately.

**Reply by Daniel-Knights** [reply]
Okay, that makes sense.

Do you think it's possible for me to make a custom tag, similar to layout or render, that would allow this feature? Perhaps by extending the layout or render class.

**Reply by harttle** [reply]
Yes, it’s definitely possible.

One way is to copy the implementation of layout, still it can be tricky to determine when to end without introducing an end tag.

Maybe a quicker way is create a tag which simply incorporates the capture & render combination. Just parse the modified template in ‘parse()’ and render it with current context in ‘render()’. Downside is the popup file is not a layout file that uses ‘{%block%}’, instead it’s a regular file uses ‘{{content}}’.

**Reply by Daniel-Knights** [reply]
Okay, thanks for the help. I'll give it a go and post back here if I figure it out
